File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        // HashSet<String> hs = new HashSet<>();
        // for (String num: phone_book) {
        //     for (int i = 1; i < num.length(); ++i) {
        //         hs.add(num.substring(0, i));
        //     }
        // }
        // for (String num: phone_book) {
        //     if (hs.contains(num)) {
        //         answer = false;
        //         break;
        //     }
        // }
        Arrays.sort(phone_book);
        for (int i = 1; i < phone_book.length; ++i) {
            if (phone_book[i].startsWith(phone_book[i - 1])) {
                answer = false;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    d = dict()
    for i in phone_book:
        s = ""
        for j in range(len(i) - 1):
            s += i[j]
            d[s] = 1
    for i in phone_book:
        if (i in d):
            answer = False
    return answer